rocks.kfs.WorkflowFromEWS/Jobs/LaunchWorkflowFromEWSAccount.cs

```diff
@@ -255,14 +372,48 @@ public virtual void Execute( IJobExecutionContext context )
                                         messages.Add( string.Format( "{0} workflow was appended.", existingWorkflow.Name ) );
                                     }
 
-                                    message.IsRead = true;
+                                    if ( markWith.Contains( ( ( int ) MarkEmailBy.Read ) ) )
+                                    {
+                                        message.IsRead = true;
+                                    }
+                                    if ( markWith.Contains( ( ( int ) MarkEmailBy.RemoveFlag ) ) )
+                                    {
+                                        message.Flag = new Flag { FlagStatus = ItemFlagStatus.NotFlagged };
+                                    }
+                                    if ( markWith.Contains( ( ( int ) MarkEmailBy.AddFlag ) ) )
+                                    {
+                                        message.Flag = new Flag { FlagStatus = ItemFlagStatus.Flagged };
+                                    }
+                                    if ( markWith.Contains( ( ( int ) MarkEmailBy.MarkComplete ) ) && message.Flag != null )
+                                    {
+                                        var flag = message.Flag;
+                                        flag.CompleteDate = RockDateTime.Now;
+                                        flag.FlagStatus = ItemFlagStatus.Complete;
+                                        message.Flag = flag;
+                                    }
+                                    // Most users will receive error "Archive mailbox is not enabled for this user."
+                                    // This archive is actual mailbox archiving, not just move to archive folder.
+                                    // Unable to test feature, requires Exchange Online Plan 2 or Exchange Online Archiving license, add note to documentation or remove?
+                                    if ( markWith.Contains( ( ( int ) MarkEmailBy.Archive ) ) )
+                                    {
+                                        archiveItemIds.Add( message.Id );
+                                    }
```

Comment:
Leave comment about archive but remove capability.
```suggestion
                                    // if ( markWith.Contains( ( ( int ) MarkEmailBy.Archive ) ) )
                                    // {
                                    //     archiveItemIds.Add( message.Id );
                                    /// }
```